src/txmempool.cpp

```diff
@@ -352,15 +352,21 @@ void CTxMemPool::removeForReorg(CChain& chain, std::function<bool(txiter)> check
     AssertLockHeld(::cs_main);
     Assume(!m_have_changeset);
 
-    setEntries txToRemove;
-    for (indexed_transaction_set::const_iterator it = mapTx.begin(); it != mapTx.end(); it++) {
-        if (check_final_and_mature(it)) txToRemove.insert(it);
+    std::vector<const TxGraph::Ref*> txToRemove;
+    {
```

Comment:
In commit "Rewrite removeForReorg to avoid using sets"

Nit: it seems this scope level is unnecessary.